cmd/mt-split-metrics-by-ttl/main.go

```diff
@@ -0,0 +1,107 @@
+	cassandraReadQueueSize       = flag.Int("cassandra-read-queue-size", 100, "max number of outstanding reads before blocking. value doesn't matter much")
+	cassandraRetries             = flag.Int("cassandra-retries", 0, "how many times to retry a query before failing it")
+	cqlProtocolVersion           = flag.Int("cql-protocol-version", 4, "cql protocol version to use")
+
+	cassandraSSL              = flag.Bool("cassandra-ssl", false, "enable SSL connection to cassandra")
+	cassandraCaPath           = flag.String("cassandra-ca-path", "/etc/raintank/ca.pem", "cassandra CA certificate path when using SSL")
+	cassandraHostVerification = flag.Bool("cassandra-host-verification", true, "host (hostname and server cert) verification when using SSL")
+
+	cassandraAuth     = flag.Bool("cassandra-auth", false, "enable cassandra authentication")
+	cassandraUsername = flag.String("cassandra-username", "cassandra", "username for authentication")
+	cassandraPassword = flag.String("cassandra-password", "cassandra", "password for authentication")
+
+	windowFactor = flag.Int("window-factor", 20, "the window factor be used when creating the metric table schema")
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "mt-split-metrics-by-ttl [flags] ttl [ttl...]")
+		fmt.Fprintln(os.Stderr)
+		fmt.Fprintln(os.Stderr, "Creates schema of metric tables split by TTLs and")
+		fmt.Fprintln(os.Stderr, "assists in migrating the data to new tables.")
+		fmt.Println("Flags:")
+		flag.PrintDefaults()
+		os.Exit(-1)
+	}
+	flag.Parse()
+
+	var ttls []uint32
+	if flag.NArg() < 1 {
+		flag.Usage()
+	}
+
+	for i := 0; i < flag.NArg(); i++ {
+		ttl, err := strconv.Atoi(flag.Arg(i))
+		if err != nil {
+			flag.Usage()
+		}
+		ttls = append(ttls, uint32(ttl))
+	}
+
+	tmpDir, err := ioutil.TempDir(os.TempDir(), *cassandraKeyspace)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to get temp dir: %s", tmpDir))
+	}
+	snapshotDir := path.Join(tmpDir, "snapshot")
+	err = os.Mkdir(snapshotDir, 0700)
+	if err != nil {
+		panic(fmt.Sprintf("Error creating directory: %s", err))
+	}
+
+	store, err := mdata.NewCassandraStore(*cassandraAddrs, *cassandraKeyspace, *cassandraConsistency, *cassandraCaPath, *cassandraUsername, *cassandraPassword, *cassandraHostSelectionPolicy, *cassandraTimeout, *cassandraReadConcurrency, *cassandraReadConcurrency, *cassandraReadQueueSize, 0, *cassandraRetries, *cqlProtocolVersion, *windowFactor, *cassandraSSL, *cassandraAuth, *cassandraHostVerification, ttls)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to instantiate cassandra: %s", err))
+	}
+	tables := store.GetTableNames()
+
+	// create directory/link structure that we need to define the future table names
+	err = os.Mkdir(path.Join(tmpDir, *cassandraKeyspace), 0700)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create directory: %s", err))
+	}
+	namedTableLinks := make([]string, 0)
```

Comment:
can make correctly sized slice and then just use `[i]` syntax to set elements instead of append ;)